### PR TITLE
Clarify TIFF/TIF images must be `uint8`

### LINKED ISSUE #22743: Validation loss becomes nan on Google Colab and Kaggle, not on local environment.
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi.
I'm training YOLOv11-seg on my computer, then migrate to Colab and Kaggle. However, on Colab and Kaggle, both of them experienced the same problem: validation loss becomes `nan`.
In both environment, I installed with:
`pip install ultralytics`
which install ultralytics 8.3.229.
Then, when I start any model, a warning appears like so, no matter pre-trained model or untrained:
```
[...]
train: Fast image access ✅ (ping: 0.0±0.0 ms, read: 522.7±802.0 MB/s, size: 1041.0 KB)
train: Scanning /root/.cache/kagglehub/datasets/hivphanphc/satellite-tiff-dirty/versions/1/tiff_full_dataset/cropped_data/train/labels.cache... 17984 images, 11642 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 17984/17984 207.8Mit/s 0.0s
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))
/usr/local/lib/python3.12/dist-packages/ultralytics/data/augment.py:679: RuntimeWarning: invalid value encountered in cast
  img4[y1a:y2a, x1a:x2a] = img[y1b:y2b, x1b:x2b]  # img4[ymin:ymax, xmin:xmax]
/usr/local/lib/python3.12/dist-packages/ultralytics/data/augment.py:679: RuntimeWarning: invalid value encountered in cast
  img4[y1a:y2a, x1a:x2a] = img[y1b:y2b, x1b:x2b]  # img4[ymin:ymax, xmin:xmax]
val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 24.6±23.8 MB/s, size: 876.8 KB)
val: Scanning /root/.cache/kagglehub/datasets/hivphanphc/satellite-tiff-dirty/versions/1/tiff_full_dataset/cropped_data/val/labels.cache... 2249 images, 1477 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 2249/2249 12.8Mit/s 0.0s
/usr/local/lib/python3.12/dist-packages/ultralytics/data/augment.py:679: RuntimeWarning: invalid value encountered in cast
  img4[y1a:y2a, x1a:x2a] = img[y1b:y2b, x1b:x2b]  # img4[ymin:ymax, xmin:xmax]
[...]
```
The warning appears several times, but not affect the training and it still runs normally with validation metrics calculated, but validation loss became `nan` in output `result.csv`.
I've tried installing `ultralytics` with colabconda, or downgrading to 8.3.223 which my computer's using with no avail.

### Additional

My computer is using Python 3.11 in a conda environment, while both environments use 3.12. My computer's GPU is GTX 960M, while I've tested L4, P100 and T4 on both environments.

### PR DESCRIPTION
**Pull request by Y-T-G**
Closes https://github.com/ultralytics/ultralytics/issues/22743

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Clarifies that multispectral TIFF images must be `uint8` in addition to having the correct shape and extension for use with Ultralytics. 🛰️

### 📊 Key Changes
- 📝 Updated the COCO8-Multispectral dataset documentation to:
  - Keep the required image layout: `(channel, height, width)`.
  - Keep supported file extensions: `.tiff` or `.tif`.
  - **Add** a new requirement that TIFF images must be stored as `uint8`.

### 🎯 Purpose & Impact
- ✅ Reduces confusion for users preparing multispectral datasets by clearly stating the expected data type.
- 🧪 Helps prevent runtime errors or unexpected behavior caused by incompatible image formats or types.
- 🚀 Makes it easier for users to correctly format and load multispectral data with Ultralytics models.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated review helper, and a Ultralytics engineer will be with you shortly. In the meantime, to ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: docs/en/datasets/detect/coco8-multispectral.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ The COCO8-Multispectral dataset is configured using a YAML file, which defines d
 
 !!! note
 
-    Prepare your TIFF images in `(channel, height, width)` order and saved with `.tiff` or `.tif` extension for use with Ultralytics:
+    Prepare your TIFF images in `(channel, height, width)` order and saved with `.tiff` or `.tif` extension for use with Ultralytics. The TIFF images must be `uint8`:
```
Comment (UltralyticsAssistant) [inline docs/en/datasets/detect/coco8-multispectral.md]: 📝 **LOW**: The added sentence reads awkwardly because the colon that introduces the code block is now attached to the “must be `uint8`” statement rather than the instruction as a whole. Consider merging the sentences so the colon clearly applies to the entire guidance.

**Suggested change:**
```suggestion
    Prepare your TIFF images in `(channel, height, width)` order, saved with `.tiff` or `.tif` extension, and ensure they are `uint8` for use with Ultralytics:
```

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Tighten the updated note so the colon continues to introduce the code snippet while communicating the new `uint8` requirement clearly.

💬 Posted 1 inline comment

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review 2

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Documentation wording change correctly clarifies the `uint8` requirement; no issues to report.

**Comment by UltralyticsAssistant** [comment]
Merged – and what a great improvement, thank you @Y-T-G! 🎉  

> “Details make perfection, and perfection is not a detail.” – Leonardo da Vinci  

Your clarification on multispectral TIFF requirements (`uint8`, layout, and extensions) is exactly that kind of high-impact detail. It will save users a lot of confusion, prevent tricky runtime issues, and make it much smoother to prepare multispectral datasets for YOLO11 and beyond.  

Really appreciate you taking the time to sharpen the docs and improve the developer experience for everyone working with Ultralytics. 🙌
